Stratis.Bitcoin.Features.Consensus/StakeItem.cs

```diff
@@ -1,13 +1,84 @@
 ﻿using NBitcoin;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.Features.Consensus
 {
-    public class StakeItem
+    /// <summary>
+    /// An object that holds block stake information.
+    /// </summary>
+    public sealed class StakeItem
     {
-        public uint256 BlockId;
-        public BlockStake BlockStake;
-        public bool InStore;
-        public long Height;
-    }
-}
+        private StakeItem() { }
+
+        /// <summary>The hash of the block.</summary>
+        public uint256 BlockHash { get; private set; }
+
+        /// <summary>The block stake.</summary>
+        public BlockStake BlockStake { get; private set; }
+
+        /// <summary>The height of the block.</summary>
+        public long? Height { get; private set; }
```

Comment:
When you use such nullable thingy, it is always beneficial to comment the null case. But is it really needed here? Wouldn't a simple long be enough?

Also, afaik everywhere Height is just integer, so why long?

Reply:
Height will never be null